lm_engine/mixins/dense/main.py

```
@@ -80,38 +50,15 @@ def _init_model(self, config: CommonConfig, **kwargs) -> None:
-
     def forward(
         self,
         input_ids: torch.Tensor | list[list[int]] | None = None,
         cache_params: GenerationCache | None = None,
-        attention_mask: torch.Tensor | None = None,
-        position_ids: torch.Tensor | list[list[int]] | None = None,
-        inputs_embeds: torch.Tensor | list[list[float]] | None = None,
-        labels: torch.Tensor | list[list[int]] | None = None,
-        use_cache: bool | None = None,
-        return_dict: bool = True,
+        attention_mask_info: AttentionMaskInfo = AttentionMaskInfo(),
```

Ruff (0.16.0)

[warning] Do not perform function call `AttentionMaskInfo` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)